src/ws.rs

```diff
@@ -1,78 +1,86 @@
-use futures_util::stream::{SplitSink, SplitStream};
-use futures_util::{SinkExt, StreamExt};
-use sysinfo::{System, SystemExt};
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use futures_util::{SinkExt, Stream, StreamExt};
 use tokio::net::TcpStream;
-use tokio_tungstenite::tungstenite::client::IntoClientRequest;
-use tokio_tungstenite::tungstenite::http::HeaderValue;
-use tokio_tungstenite::{tungstenite::Message, MaybeTlsStream, WebSocketStream};
+use tokio_tungstenite::tungstenite::Error;
+use tokio_tungstenite::{
+    tungstenite::client::IntoClientRequest, tungstenite::http::HeaderValue, tungstenite::Message,
+    Connector, MaybeTlsStream, WebSocketStream,
+};
 
+use crate::model::ws::{deserialize::DeEvent, LcuEvent, LcuSubscriptionType, LcuWebsocketError};
 use crate::utils::process_info::*;
 
-pub struct WSClient {
-    pub write: SplitSink<WebSocketStream<MaybeTlsStream<TcpStream>>, Message>,
-    pub read: SplitStream<WebSocketStream<MaybeTlsStream<TcpStream>>>,
-}
-
-type Error = Box<dyn std::error::Error>;
-
-#[derive(PartialEq)]
-pub enum Events {
-    Json,
-    Lcds,
-    Callback,
-    None,
-}
+pub struct WSClient(WebSocketStream<MaybeTlsStream<TcpStream>>);
 
 impl WSClient {
-    pub async fn connect(events: Events) -> Result<Self, Error> {
-        let mut sys = System::new_all();
-        sys.refresh_all();
-        let process = find_process(&sys)?;
-        let args = extract_info(process)?;
-        let auth_token = encode_token(&args.0);
+    pub async fn connect() -> Result<Self, LcuWebsocketError> {
+        let (auth_token, port) = get_auth_info().map_err(|_| LcuWebsocketError::LcuNotAvailable)?;
```

Comment:
Does `.unwrap_or(Err(LcuWebsocketError::LcuNotAvailable)?;` work? It would be better for readability